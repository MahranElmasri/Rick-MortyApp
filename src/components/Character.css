/* Cards */
.card {
  margin-left: auto;
  margin-right: auto;
  width: max-content;
  padding: 1rem;
  border: rgb(126, 114, 114) 1px solid;
  margin: 0.7rem 0;
  border-radius: 20px;
  box-shadow: 0px 3px 9px rgba(134, 126, 126, 0.42);
}

.character-img {
  max-width: 300px;
  border-radius: 5px;
}
