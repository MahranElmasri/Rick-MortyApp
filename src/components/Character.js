import { Link } from "react-router-dom";
import React from "react";
import Spinner from "./layout/Spinner";
import PropTypes from "prop-types";
import "./Character.css";

export default function Character({ char, loading }) {
  const { id, image, name, gender, species, status, created } = char;
  return (
    <div data-testid="char" className="char">
      <ul className="app">
        <li className=" enter card" key={id}>
          <Link to={`/details/${id}`}>
            {loading ? (
              <Spinner />
            ) : (
              <img className="character-img" src={image} alt="charImage" />
            )}
            <div className="center text-light ">{name}</div>
            <div className="center text-light">{species}</div>
            <div className="center text-light">{gender}</div>
            <div className="center text-light">{created.split("T")[0]}</div>
            <div
              className={
                status === "Dead"
                  ? "badge-danger"
                  : status === "Alive"
                  ? "badge-success"
                  : "badge"
              }
            >
              {status}
            </div>
          </Link>
        </li>
      </ul>
    </div>
  );
}

Character.propTypes = {
  char: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired,
};
