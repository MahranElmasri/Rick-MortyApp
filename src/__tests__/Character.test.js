import Character from "../components/Character";

it("render correctlty", () => {
  const char = {
    id: 1,
    image: "",
    name: "Rick Sanchex",
    gender: "Male",
    species: "Human",
    status: "Alive",
    created: "2017-11-04",
  };
  const wrapper = shallow(<Character char={char} loading={false} />);
  expect(wrapper).toMatchSnapshot();
});
