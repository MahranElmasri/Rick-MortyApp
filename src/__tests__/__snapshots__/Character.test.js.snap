// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctlty 1`] = `
<div
  className="char"
  data-testid="char"
>
  <ul
    className="app"
  >
    <li
      className=" enter card"
      key="1"
    >
      <Link
        to="/details/1"
      >
        <img
          alt="charImage"
          className="character-img"
          src=""
        />
        <div
          className="center text-light "
        >
          Rick Sanchex
        </div>
        <div
          className="center text-light"
        >
          Human
        </div>
        <div
          className="center text-light"
        >
          Male
        </div>
        <div
          className="center text-light"
        >
          2017-11-04
        </div>
        <div
          className="badge-success"
        >
          Alive
        </div>
      </Link>
    </li>
  </ul>
</div>
`;
